src/main/java/com/toptrumps/core/engine/GameEventListener.java

```diff
@@ -12,6 +12,8 @@
     Attribute onRequestSelection(final Card card);
     void onAttributeSelected(final Player activePlayer);
     void onRoundEnd(final RoundOutcome outcome);
+    void onRoundEnd(final RoundOutcome outcome, final Player eliminatedPlayer);
+    void onRoundEnd(final RoundOutcome outcome, final ArrayList<Player> eliminatedPlayers);
```

Comment:
These are supposed to be life cycle methods. It's very confusing to have overloads in this case! We should keep online one "onRoundEnd" and pass to it everything we might need. If we have +3 parameters, then it's time to create an object.